### DISCUSSION TITLE
Performance degradation due to PR #4383 to reduce no-op wakeups in multi-threaded scheduler

### DISCUSSION THREAD

**Discussion by manaswini05** [tokio-rs/tokio#4869] [Q&A]
Hey! I recently tried upgrading tokio from v1.13.1 to v1.16.1 and observed quite a performance degradation in a real-life service. Reverting this PR - https://github.com/tokio-rs/tokio/pull/4383 - seems to fix it. We believe there is reduced concurrency as our CPU usage drops and our response times increase significantly. Were there any rammifications observed with the mentioned PR or any ideas here? Thank you!

**Comment by Darksonn** [comment]
This is the first time I hear about that PR causing a performance degradation, but apparently you are not the only one.

Can you please open an issue instead of a discussion? Discussions are for general help with Tokio, whereas issues are for bugs.

**Reply by manaswini05** [reply]
Yeap sure!
